### PR TITLE
Pr2 branch

### PR DESCRIPTION
**Pull request by araslav**

### CODE REVIEW

**Comment by araslav** [comment]
HW for Spring Boot Web theme

File: pom.xml [hunk 1]
```diff
@@ -72,6 +76,7 @@
 					<consoleOutput>true</consoleOutput>
 					<failsOnError>true</failsOnError>
 					<linkXRef>false</linkXRef>
+					<sourceDirectories>src</sourceDirectories>
```
Comment (sokimaaa) [inline pom.xml]: wrong way of using sourceDirectories attribute. 
add separately 
src/main/java and src/test/java folders

```suggestion
					<sourceDirectories><sourceDirectory>src/main/java</sourceDirectory></sourceDirectories>
```

File: src/main/java/book/store/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api")
+public class BookController {
+    private final BookService bookService;
+
+    @PostMapping("/books")
+    public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
```
Comment (sokimaaa) [inline src/main/java/book/store/controller/BookController.java:22]: see any validation here

File: pom.xml [hunk 3]
```diff
@@ -42,6 +42,10 @@
 			<artifactId>spring-boot-starter-data-jpa</artifactId>
 			<version>3.2.1</version>
 		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-validation</artifactId>
+		</dependency>
```
Comment (sokimaaa) [inline pom.xml]: see any usage of this. 
dependencies that not using should be removed
```suggestion
```

File: src/main/java/book/store/dto/request/CreateBookRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package book.store.dto.request;
+
+import jakarta.validation.constraints.NotNull;
+import java.math.BigDecimal;
+import lombok.Data;
+import org.springframework.stereotype.Component;
+
+@Data
+@Component
```
Comment (sokimaaa) [inline src/main/java/book/store/dto/request/CreateBookRequestDto.java]: never use annotations if you dont understand how they work
```suggestion
```

File: src/main/java/book/store/dto/request/CreateBookRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+public class CreateBookRequestDto {
+    @NotNull
+    private String title;
+    @NotNull
+    private String author;
+    @NotNull
+    private String isbn;
+    @NotNull
```
Comment (sokimaaa) [inline src/main/java/book/store/dto/request/CreateBookRequestDto.java]: never use annotations if you dont understand how they work



```suggestion
    private String title;
    private String author;
    private String isbn;
```

File: src/main/java/book/store/dto/response/BookDto.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package book.store.dto.response;
+
+import java.math.BigDecimal;
+import lombok.Data;
+import org.springframework.stereotype.Component;
+
+@Data
+@Component
```
Comment (sokimaaa) [inline src/main/java/book/store/dto/response/BookDto.java]: never use annotations if you dont understand how they work
```suggestion
```

File: src/main/java/book/store/mapper/request/BookRequestDtoMapper.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+import book.store.dto.request.CreateBookRequestDto;
+import book.store.mapper.RequestDtoMapper;
+import book.store.model.Book;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BookRequestDtoMapper implements RequestDtoMapper<CreateBookRequestDto, Book> {
+    @Override
+    public Book toModel(CreateBookRequestDto dto) {
```
Comment (sokimaaa) [inline src/main/java/book/store/mapper/request/BookRequestDtoMapper.java]: image

File: src/main/java/book/store/mapper/response/BookResponseDtoMapper.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+import book.store.dto.response.BookDto;
+import book.store.mapper.ResponseDtoMapper;
+import book.store.model.Book;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BookResponseDtoMapper implements ResponseDtoMapper<Book, BookDto> {
+    @Override
+    public BookDto toDto(Book model) {
+        BookDto bookDto = new BookDto();
```
Comment (sokimaaa) [inline src/main/java/book/store/mapper/response/BookResponseDtoMapper.java]: image

File: src/main/java/book/store/service/impl/BookServiceImpl.java [hunk 9]
```diff
@@ -11,14 +16,26 @@
...
+                .save(bookRequestDtoMapper.toModel(requestDto)));
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto getBookById(Long id) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Don't find book with id " + id));
+        return bookResponseDtoMapper.toDto(book);
```
Comment (sokimaaa) [inline src/main/java/book/store/service/impl/BookServiceImpl.java]: inline expression using Optional#map method
```suggestion
        Book book = bookRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException("Don't find book with id " + id));
        return bookResponseDtoMapper.toDto(book);
```

File: src/test/java/book/store/BooksStoreApplicationTests.java [hunk 10]
```diff
@@ -4,10 +4,8 @@
 import org.springframework.boot.test.context.SpringBootTest;
 
 @SpringBootTest
-class BooksStoreApplicationTests {
-
-	@Test
-	void contextLoads() {
-	}
-
+public class BooksStoreApplicationTests {
```
Comment (sokimaaa) [inline src/test/java/book/store/BooksStoreApplicationTests.java]:
```suggestion
class BooksStoreApplicationTests {
```

File: src/main/java/book/store/dto/request/CreateBookRequestDto.java (hunk 4 above)
Comment (araslav) [reply src/main/java/book/store/dto/request/CreateBookRequestDto.java]: @sokimaaa I know that this annotation mark the beans in Spring, this annotation is the "parent" for @Service, @Repository, @Controller annotations. 
This annotation used in video about DTO for DTO.
Please, explain why I must not use @Component and what annotation I have to use instead last one, I didn't find this information.

File: pom.xml (hunk 3 above)
Comment (araslav) [reply pom.xml]: @sokimaaa I am using validation in package book.store.dto.request, class CreateBookRequestDto.
Is this solution doesn't correct?

File: src/main/java/book/store/dto/request/CreateBookRequestDto.java (hunk 5 above)
Comment (araslav) [reply src/main/java/book/store/dto/request/CreateBookRequestDto.java]: @sokimaaa 
I am using this annotation for receiving not null values from requests to the server in the BookController:
@PostMapping("/books")
    public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
        return bookService.createBook(requestDto);
    }
Is this solution doesn't correct?

File: src/main/java/book/store/controller/BookController.java [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api")
```
Comment (Rommelua) [inline src/main/java/book/store/controller/BookController.java]:
```suggestion
@RequestMapping("/api/books")
```

File: src/main/java/book/store/controller/BookController.java [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+public class BookController {
+    private final BookService bookService;
+
+    @PostMapping("/books")
```
Comment (Rommelua) [inline src/main/java/book/store/controller/BookController.java]:
```suggestion
    @PostMapping
```

Comment (araslav) [reply src/main/java/book/store/controller/BookController.java]: @Rommelua Done

File: src/main/java/book/store/controller/BookController.java [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+        return bookService.getBookById(id);
+    }
+
+    @GetMapping
+    public List<BookDto> getAll() {
+        return bookService.getAll();
+    }
+}
+
+
```
Comment (koval-kn) [inline src/main/java/book/store/controller/BookController.java:37]:
```suggestion
```

**Review by koval-kn — APPROVED** [review]

**Pull request merged**